include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -579,7 +515,7 @@ class operational_domain_impl
         // add the neighbors of each operational point to the queue
         for (const auto& [param_point, status] : op_domain.get_domain())
         {
-            if (status == operational_status::OPERATIONAL)
+            if (std::get<0>(status) == operational_status::OPERATIONAL)
```

Comment:
This only works if the first entry in the tuple of mapped values in `OpDomain` is of type `operational_status`. This is not guaranteed if you're calling this function, or? As per usual, there are multiple possible solutions to this: We could for example restrict calls via `static_assert`s.

Reply:
I mean, we already use ``static_assert`` in ``operational_domain_flood_fill``, for example, where we restrict that scenario.